### PR TITLE
Updates to map component

### PR DESCRIPTION
**Pull request by paylakatel**
This PR:
- brings back feature counts so we can display them on the map
- changes how we render the map from leaflet to mapboxGL to improve performance
- adds a heatmap layer on zoom out
- makes sure the pop-ups display information about all the crashes if a user selects multiple points
- sort of deals with timezones by adding 5 hours to our dates - a method that may need to be made more robust in the future

### CODE REVIEW

File: components/Map.js [hunk 1]
```diff
@@ -1,198 +1,513 @@
...
-const { basemapLayer, tiledMapLayer, featureLayer } = process.browser
-  ? require('esri-leaflet')
-  : {};
-
-const basemap_url =
-  'https://awsgeo.boston.gov/arcgis/rest/services/Basemaps/BostonCityBasemap_WM/MapServer';
+const mapboxgl = process.browser ? require('mapbox-gl') : null;
+// Despite using mapboxgl to render the map, we still use esri-leaflet to
+// query to the layer
+const { featureLayer } = process.browser ? require('esri-leaflet') : {};
```
Comment (fionawhim) [inline components/Map.js:10]: Add a `TODO` to remove that dependency and just AJAX the info in.

File: components/Map.js [hunk 2]
```diff
@@ -1,198 +1,513 @@
...
+            type: 'raster',
+            tiles: [
+              'https://services.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Base/MapServer/tile/{z}/{y}/{x}',
+            ],
+            tileSize: 256,
+          },
+          'cob-basemap': {
+            type: 'raster',
+            tiles: [
+              'https://awsgeo.boston.gov/arcgis/rest/services/Basemaps/BostonCityBasemap_WM/MapServer/tile/{z}/{y}/{x}',
```
Comment (fionawhim) [inline components/Map.js:58]: cool!

File: components/Map.js [hunk 3]
```diff
@@ -1,198 +1,513 @@
...
-      },
-      onEachFeature: this.bindPopUp,
     });
 
-    this.fatalityFeatureLayer.setWhere(allModesSelected);
+    // Bind pop-ups for crash and fatality points
+    this.map.on('click', e => {
+      // Crash and fatalitiy points are often stacked on top of eachother
+      // in the data. We want the pop-ups to show information about all
+      // the crashes that occured in the clicked location.
```
Comment (fionawhim) [inline components/Map.js:308]: fancy!

File: components/Map.js [hunk 4]
```diff
@@ -1,198 +1,513 @@
...
+          feature.properties.mode_type,
+          feature.properties[dateField],
+        ])
+        // We sort crashes listed in the pop-up by most-recent
+        // to least recent
+        .sort((a, b) => {
+          return a[1] > b[1] ? -1 : 1;
+        });
+
+      new mapboxgl.Popup({ closeOnClickboolean: true })
```
Comment (fionawhim) [inline components/Map.js]: Should this just be `closeOnClick: true`?

File: components/Map.js (hunk 1 above)
Comment (paylakatel) [reply components/Map.js:10]: added here: https://github.com/CityOfBoston/vision-zero/issues/25. Could be a good debt bash item for next week

File: components/Map.js (hunk 4 above)
Comment (paylakatel) [reply components/Map.js]: lol yes - thank you

**Review by fionawhim — APPROVED** [review]
LGTM!

**Pull request merged**